discord/abc.py

```diff
@@ -745,6 +752,9 @@ def permissions_for(self, obj: Union[Member, Role], /) -> Permissions:
             return Permissions.all()
 
         default = self.guild.default_role
+        if default is None:
+            return Permissions._user_installed_permissions(in_guild=True)
```

Comment:
Doesn't this need a check to ensure it's the bot user? This could return incorrect permissions if one uses `permissions_for` on a user other than the bot.

Reply:
That is a good point.
What do you think would be the most sensible way of returning user permissions? Just 0? Also, how do I access the bot/bot ID in this method?

Reply:
You can get the bot id from `channel._state.self_id`. I don't have any better ideas other than 0.

Reply:
Cheers, that should be implemented now